src/foundry/roll.d.ts

```diff
@@ -271,7 +271,8 @@ declare class Roll<D extends object = {}> {
    * console.log(r.total);  // 11
    * ```
    */
-  evaluate({ minimize, maximize, async }?: Partial<Roll.Options>): this | Promise<this>;
+  evaluate({ minimize, maximize, async }?: Partial<Roll.Options & { async: false }>): this;
+  evaluate({ minimize, maximize, async }?: Partial<Roll.Options & { async: true }>): Promise<this>;
```

Comment:
You should also add an indeterminate variant, because not all booleans are known at compile time. Or in other words, leave the original version in, and just put it at the end of the trio.